app/src/main/java/com/stevesoltys/seedvault/restore/RestoreViewModel.kt

```diff
@@ -307,7 +317,28 @@ internal class RestoreViewModel(
          *   the current device. If no applicable datasets exist, restoreSets will be null.
          */
         override fun restoreSetsAvailable(restoreSets: Array<out RestoreSet>?) {
-            startRestore(this)
+            // this gets executed after we got the restore sets
+            // now we can start the restore of all available packages
+            restoreNextPackages()
+        }
+
+        /**
+         * Restore the next chunk of packages.
+         *
+         * We need to restore in chunks, otherwise [BackupTransport.startRestore] in the
+         * framework's [PerformUnifiedRestoreTask] may fail due to an oversize Binder
+         * transaction, causing the entire restoration to fail.
+         */
+        private fun restoreNextPackages() {
+            val chunkSize = (packageIndex + PACKAGES_PER_CHUNK).coerceAtMost(packages.size)
+            val packageChunk = packages.subList(packageIndex, chunkSize).toTypedArray()
+            packageIndex += packageChunk.size
+
+            val result = session.restorePackages(token, this, packageChunk, monitor)
+
+            if (result != 0) {
+                Log.e(TAG, "restorePackages() returned non-zero value")
```

Comment:
This is nothing we need to surface to the user? Should we also log the result?

Reply:
Error handling will go in the `restoreFinished` function which should always be called after this. Not sure if we even need to log this tbh.